geo/src/algorithm/concave_hull.rs

```diff
@@ -0,0 +1,464 @@
+    };
+    let mut candidates = interior_coords_tree
+        .locate_within_distance(centroid_coord, search_dist)
+        .peekable();
+    let peek = candidates.peek();
+    match peek {
+        None => None,
+        Some(&point) => {
+            let closest_point =
+                candidates.fold(Point::new(point.x, point.y), |acc_point, candidate| {
+                    let candidate_point = Point::new(candidate.x, candidate.y);
+                    if line.euclidean_distance(&acc_point)
+                        > line.euclidean_distance(&candidate_point)
+                    {
+                        candidate_point
+                    } else {
+                        acc_point
+                    }
+                });
+            let mut edges_nearby_point = line_tree
+                .locate_within_distance(closest_point, search_dist)
+                .peekable();
+            let peeked_edge = edges_nearby_point.peek();
+            let closest_edge_option = match peeked_edge {
+                None => None,
+                Some(&edge) => Some(edges_nearby_point.fold(*edge, |acc, candidate| {
+                    if closest_point.euclidean_distance(&acc)
+                        > closest_point.euclidean_distance(candidate)
+                    {
+                        *candidate
+                    } else {
+                        acc
+                    }
+                })),
+            };
+            let decision_distance = partial_min(
+                closest_point.euclidean_distance(&line.start_point()),
+                closest_point.euclidean_distance(&line.end_point()),
+            );
+            if let Some(closest_edge) = closest_edge_option {
+                let far_enough = edge_length / decision_distance > concavity;
+                let are_edges_equal = closest_edge == line;
+                if far_enough && are_edges_equal {
+                    Some(Coordinate {
+                        x: closest_point.x(),
+                        y: closest_point.y(),
+                    })
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
+        }
+    }
+}
+
+// This takes significant inspiration from:
+// https://github.com/mapbox/concaveman/blob/54838e1/index.js#L11
+fn concave_hull<T>(coords: &mut [Coordinate<T>], concavity: T) -> LineString<T>
+where
+    T: Float + RTreeNum + HasKernel,
+{
+    let hull = qhull::quick_hull(coords);
+
+    if coords.len() < 4 {
+        return hull;
+    }
+
+    //Get points in overall dataset that aren't on the exterior linestring of the hull
+    let hull_tree: RTree<Coordinate<T>> = RTree::bulk_load(hull.clone().0);
+
+    let interior_coords: Vec<Coordinate<T>> = coords
+        .iter()
+        .filter_map(|coord| {
+            if !hull_tree.contains(coord) {
+                Some(*coord)
+            } else {
+                None
+            }
+        })
```

Comment:
i think this is the same?

```suggestion
        .filter(|coord| !hull_tree.contains(coord))
```

Reply:
Thanks for catching that! I've made a change to account for this.